File: frontend/de/unika/ipd/grgen/ast/decl/AbstractMemberDeclNode.java
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 5.0
 * Copyright (C) 2003-2020 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3 (see LICENSE.txt included in the packaging of this file)
 * www.grgen.net
 */

package de.unika.ipd.grgen.ast.decl;

import de.unika.ipd.grgen.ast.IdentNode;
import de.unika.ipd.grgen.ast.type.BasicTypeNode;

/**
 * A compound type member declaration that is abstract, i.e. has no type defined yet, but just a member name.
 */
public class AbstractMemberDeclNode extends MemberDeclNode
{
	static {
		setName(AbstractMemberDeclNode.class, "abstract member declaration");
	}

	/**
	 * @param n Identifier which declared the member.
	 * @param t Type with which the member was declared.
	 */
	public AbstractMemberDeclNode(IdentNode n, boolean isConst)
	{
		super(n, BasicTypeNode.voidType, isConst);
	}
}

File: frontend/de/unika/ipd/grgen/ast/model/decl/AbstractMemberDeclNode.cs
/*
 * GrGen: graph rewrite generator tool -- release GrGen.NET 8.1
 * Copyright (C) 2003-2026 Universitaet Karlsruhe, Institut fuer Programmstrukturen und Datenorganisation, LS Goos; and free programmers
 * licensed under LGPL v3, some components/parts use different licenses (see LICENSE.txt included in the packaging of this file)
 * www.grgen.de / www.grgen.net
 */

namespace de.unika.ipd.grgen.ast.model.decl
{
	using IdentNode = de.unika.ipd.grgen.ast.IdentNode;
	using BasicTypeNode = de.unika.ipd.grgen.ast.type.basic.BasicTypeNode;

	/// <summary>
	/// A compound type member declaration that is abstract, i.e. has no type defined yet, but just a member name.
	/// </summary>
	public class AbstractMemberDeclNode : MemberDeclNode
	{
		static AbstractMemberDeclNode()
		{
			SetClassName(typeof(AbstractMemberDeclNode), "abstract member declaration");
		}

		/// <param name="n"> Identifier which declared the member. </param>
		/// <param name="t"> Type with which the member was declared. </param>
		public AbstractMemberDeclNode(IdentNode n, bool isConst)
			: base(n, BasicTypeNode.voidType, isConst)
		{
		}
	}

}
